bliss.js

```diff
@@ -95,6 +129,19 @@ extend($, {
 		return $.set(document.createElement(tag || "div"), o);
 	},
 
+	hooks: {
+		add: function (name, callback) {
+			this[name] = this[name] || [];
+			this[name].push(callback);
+		},
+
+		run: function (name, env) {
+			(this[name] || []).forEach(function(callback) {
+				callback(env);
+			});
+		}
+	},
+
```

Comment:
This is the odd bit of code that was added. I had removed that, since hooks are now an instance of `$.Hooks`.